pty/src/win/mod.rs

```diff
@@ -85,7 +89,9 @@ impl std::future::Future for WinChild {
             Ok(None) => {
                 struct PassRawHandleToWaiterThread(pub RawHandle);
                 unsafe impl Send for PassRawHandleToWaiterThread {}
-                let handle = PassRawHandleToWaiterThread(self.proc.as_raw_handle());
+
+                let proc = self.proc.lock().unwrap();
```

Comment:
It'd probably be better (wrt. unexpected deadlock/blocking) to constrain the scope of this lock, similar to the first suggestion I made above.

This `PassRawHandleToWaiterThread` construct is a (pre-existing) little wonky thing.  Similar to above it is safer to `try_clone`, but note that `poll` could be called relatively more frequently, so it is potentially more likely to fail to clone the handle, but also: this code is already spawning a new thread for each call through this code path, which is already a bit gross, but unavoidable because each call has its own `waker`, so I don't know if it really makes anything worse to add a `try_clone` in here.

Reply:
That makes sense to me, and like you said, if the `try_clone` only happens in case of lack of resources then there are probably other things to worry about in this case.